pywinauto/controls/uia_controls.py

```diff
@@ -926,9 +928,88 @@ def texts(self):
     #----------------------------------------------------------------
     def button_count(self):
         """Return a number of buttons on the ToolBar"""
-        return len(self.children())
+        children_list = self.children()
+        if not children_list and self.element_info.handle is not None:
+            btn_count = common_controls.ToolbarWrapper(self.element_info.handle).button_count()
+            return btn_count
+        else:
+            return len(children_list)
 
     # ----------------------------------------------------------------
+    def _menu_item_exists(self, btn_x_coord, btn_y_coord):
```

Comment:
This method could be in-place function inside method `_up_down_menu_items` because it's used only there and it doesn't require `self` object.